pkg/odo/cli/component/devfile.go

```diff
@@ -27,11 +37,40 @@ func (po *PushOptions) DevfilePush() (err error) {
 		return err
 	}
 
-	// Write back devfile yaml
-	err = devObj.WriteYamlDevfile()
+	componentName, err := getComponentName()
+	if err != nil {
+		return err
+	}
+
+	spinner := log.SpinnerNoSpin(fmt.Sprintf("Push devfile component %s", componentName))
+	defer spinner.End(false)
+
+	devfileHandler, err := adapters.NewPlatformAdapter(componentName, devObj)
 	if err != nil {
 		return err
 	}
 
-	return nil
+	err = devfileHandler.Start()
+	if err != nil {
+		log.Errorf(
+			"Failed to start component with name %s.\nError: %v",
+			componentName,
+			err,
+		)
+		os.Exit(1)
+	}
+
+	spinner.End(true)
+	return
+}
+
+func getComponentName() (string, error) {
```

Comment:
Perhaps, a description of this function would help the reader understand what this function does.

Reply:
Done.